### PR TITLE
Rckirby/gently

### PR DESCRIPTION
**Pull request by rckirby**
Referees (and occasionally users) have asked for support for DIRKs.  For linear problems, existing demos DTRT with just using solver parameters (although they use too much memory building a triangular system for all the stages rather than one system that's reused for all stages) but it's hard to steer nonlinear problems to solve for one stage at a time.

### CODE REVIEW

**Comment by rckirby** [comment]
After opening the PR, it seems like a better option for the interface might be to let `TimeStepper` take an optional kwarg `dirk=False` that, if true, returns a `DIRKTimeStepper` rather than requiring the user to invoke the constructor.  It would be more similar to the other options we've put in.

File: irksome/tools.py [hunk 1]
```diff
@@ -9,6 +9,9 @@
 from irksome.deriv import TimeDerivative
 
 
+vecconst = numpy.vectorize(Constant)
```
Comment (ScottMacLachlan) [inline irksome/tools.py:12]: This doesn't seem to be being used, but duplicates a line from `stage.py`.  Am I missing something or is this truly orphaned?

**Review by ScottMacLachlan — COMMENTED** [review]
I'm committing a couple of small cleanups, but otherwise this looks good to me.

File: irksome/tools.py (hunk 1 above)
Comment (rckirby) [reply irksome/tools.py:12]: At some point I was planning to stick `vecconst` in `tools.py` but this line is currently orphaned.

**Pull request merged**